### PR TITLE
Remove FLOPS based mission method

### PR DESCRIPTION
**Pull request by johnjasa**
### Summary

I've modified all run scripts, tests, benchmarks, and docs to use the `simple` mission method and removed the `FLOPS` mission method.
This is probably the crux of my work this jog.
This PR dramatically simplifies the FLOPS-based mission logic and run scripts.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Comment by Kenneth-T-Moore** [comment]
I am a little nervous about completely removing the old method. We might want it to be able to build off of legacy results using the new aviary, or we might want to do some testing with it.

**Comment by johnjasa** [comment]
> I am a little nervous about completely removing the old method. We might want it to be able to build off of legacy results using the new aviary, or we might want to do some testing with it.

Can you tell me more about the use cases you'd want to keep it around for? The underlying physics of the simple height energy equations of motion are the same as in FLOPS. Would you ever want new users to use the older mission method or just keep it around for comparisons?

**Comment by johnjasa** [comment]
Ken and I had a great conversation about this and established this path forward:

- Irian will convert the MBSA&E models to use Aviary 0.9.1+
- Ken and John will help convert this to the simple mission
- Once MBSA&E is good with the new setup and results, we can merge in this PR

File: aviary/docs/getting_started/onboarding_level1.ipynb
Comment (jkirk5) [inline aviary/docs/getting_started/onboarding_level1.ipynb:1]: Why change height_energy to simple? And phase info paths are wrong. I guess this whole file needs to get updated, seems like it is very out-of-date on how to set up mission & phase info

File: aviary/interface/methods_for_level2.py [hunk 1]
```diff
@@ -694,7 +691,7 @@ def _add_groundroll_eq_constraint(self, phase):
             promotes_inputs=["t_init_gear", "t_init_flaps"],
         )
 
-    def _get_height_energy_phase(self, phase_name, phase_idx):
+    def _get_simple_phase(self, phase_name, phase_idx):
```
Comment (jkirk5) [inline aviary/interface/methods_for_level2.py:694]: Why change from height_energy to simple?

File: aviary/interface/methods_for_level2.py [hunk 2]
```diff
@@ -1074,10 +1050,10 @@ def add_subsystem_timeseries_outputs(phase, phase_name):
                     if phase_name == 'ascent':
                         self._add_groundroll_eq_constraint(phase)
 
-        elif self.mission_method is HEIGHT_ENERGY or self.mission_method is SIMPLE:
+        elif self.mission_method is HEIGHT_ENERGY:
             for phase_idx, phase_name in enumerate(phases):
                 phase = traj.add_phase(
-                    phase_name, self._get_height_energy_phase(phase_name, phase_idx))
+                    phase_name, self._get_simple_phase(phase_name, phase_idx))
```
Comment (jkirk5) [inline aviary/interface/methods_for_level2.py:1056]: same question here, why bring back simple

File: aviary/interface/test/test_check_phase_info.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 
 class TestCheckInputs(unittest.TestCase):
-    def test_correct_input_flops(self):
+    def test_correct_input_simple(self):
```
Comment (jkirk5) [inline aviary/interface/test/test_check_phase_info.py:14]: More simple! FLOPS -> height_energy

File: aviary/interface/test/test_check_phase_info.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
...
         # This should pass without any issue as it's the same valid dict as before
         self.assertTrue(check_phase_info(phase_info_flops, mission_method=HEIGHT_ENERGY))
```
Comment (jkirk5) [inline aviary/interface/test/test_check_phase_info.py:16]: flops & gasp in this file should get changed to height_energy and 2dof

File: aviary/interface/test/test_phase_info.py [hunk 5]
```diff
@@ -80,11 +76,11 @@ def _test_phase_info_dict(self, phase_info_dict, name):
                         raise RuntimeError(
                             f'value mismatch ({key}): {lhs_value} != {rhs_value}')
 
-    def test_default_phase_flops(self):
-        """Tests the roundtrip conversion for default_phase_info.flops"""
+    def test_default_phase_simple(self):
```
Comment (jkirk5) [inline aviary/interface/test/test_phase_info.py:79]: Just marking all the remaining instances of simple

File: aviary/interface/utils/check_phase_info.py [hunk 6]
```diff
@@ -17,48 +16,25 @@ def check_phase_info(phase_info, mission_method):
...
+        'throttle_enforcement': str,
+        'constrain_final': bool,
+        'fix_duration': bool,
+        'initial_bounds': tuple,
+        'duration_bounds': tuple,
     }
 
     # Combine common and phase-specific entries
-    phase_keys_flops = {
+    phase_keys_simple = {
```
Comment (jkirk5) [inline aviary/interface/utils/check_phase_info.py:36]: here

File: aviary/mission/flops_based/ode/mission_EOM.py [hunk 7]
```diff
@@ -14,21 +15,18 @@ def initialize(self):
...
-        self.add_subsystem(name=Dynamic.Mission.ALTITUDE_RATE,
-                           subsys=AltitudeRate(num_nodes=nn),
-                           promotes_inputs=[Dynamic.Mission.SPECIFIC_ENERGY_RATE,
-                                            Dynamic.Mission.VELOCITY_RATE, Dynamic.Mission.VELOCITY],
-                           promotes_outputs=[Dynamic.Mission.ALTITUDE_RATE])
+        self.add_subsystem(name='required_thrust',
+                           subsys=RequiredThrust(num_nodes=nn),
+                           promotes_inputs=[Dynamic.Mission.DRAG, Dynamic.Mission.ALTITUDE_RATE,
+                                            Dynamic.Mission.VELOCITY, Dynamic.Mission.VELOCITY_RATE, Dynamic.Mission.MASS],
+                           promotes_outputs=['T_required'])
```
Comment (jkirk5) [inline aviary/mission/flops_based/ode/mission_EOM.py:22]: verbose variable name please! `thrust_required`

File: aviary/mission/flops_based/ode/mission_ODE.py [hunk 8]
```diff
@@ -42,6 +39,10 @@ def initialize(self):
         self.options.declare(
             'use_actual_takeoff_mass', default=False,
             desc='flag to use actual takeoff mass in the climb phase, otherwise assume 100 kg fuel burn')
+        self.options.declare(
```
Comment (jkirk5) [inline aviary/mission/flops_based/ode/mission_ODE.py:42]: add `values` argument to define what options are acceptable

File: aviary/models/test_aircraft/aircraft_for_bench_GwFm_simple.csv
Comment (jkirk5) [inline aviary/models/test_aircraft/aircraft_for_bench_GwFm_simple.csv:1]: This file also no longer needs to exist

File: aviary/models/test_aircraft/aircraft_for_bench_FwFm_simple.csv
Comment (jkirk5) [inline aviary/models/test_aircraft/aircraft_for_bench_FwFm_simple.csv:1]: You can delete this file - it should be an exact copy of FwFm.csv now

File: aviary/validation_cases/benchmark_tests/test_0_iters.py [hunk 9]
```diff
@@ -45,9 +45,9 @@ def test_gasp_zero_iters(self):
 class HEZeroItersTestCase(BaseProblemPhaseTestCase):
 
     @require_pyoptsparse(optimizer="IPOPT")
-    def test_height_energy_zero_iters(self):
-        local_phase_info = deepcopy(height_energy_phase_info)
-        self.build_and_run_problem('models/test_aircraft/aircraft_for_bench_FwFm.csv',
+    def test_simple_zero_iters(self):
```
Comment (jkirk5) [inline aviary/validation_cases/benchmark_tests/test_0_iters.py:48]: here too

File: aviary/validation_cases/benchmark_tests/test_0_iters.py [hunk 10]
```diff
@@ -5,7 +5,7 @@
 
 from aviary.interface.methods_for_level2 import AviaryProblem
 from aviary.interface.default_phase_info.two_dof import phase_info as two_dof_phase_info
-from aviary.interface.default_phase_info.height_energy import phase_info as height_energy_phase_info
+from aviary.interface.default_phase_info.height_energy import phase_info as simple_phase_info
```
Comment (jkirk5) [inline aviary/validation_cases/benchmark_tests/test_0_iters.py:8]: more height_energy reverted to simple

File: aviary/validation_cases/benchmark_tests/test_subsystems_within_a_mission.py [hunk 11]
```diff
@@ -15,48 +14,36 @@
...
                 'include_takeoff': False,
                 'external_subsystems': [ArrayGuessSubsystemBuilder(), AdditionalArrayGuessSubsystemBuilder()],
-                'optimize_mass': False,
+                'optimize_mass': True,
             },
             'cruise': {
-                'subsystem_options': {
-                    'core_aerodynamics': {'method': 'solved_alpha', 'aero_data': aero_data, 'training_data': False}
-                },
+                "subsystem_options": {"core_aerodynamics": {"method": "computed"}},
```
Comment (jkirk5) [inline aviary/validation_cases/benchmark_tests/test_subsystems_within_a_mission.py:24]: Why was aero method changed? It should be mission agnostic (and if it is somehow not, then we need a bug report!)

File: aviary/variable_info/enums.py [hunk 12]
```diff
@@ -57,10 +57,9 @@ class EquationsOfMotion(Enum):
     """
     Available equations of motion for use during mission analysis
     """
-    HEIGHT_ENERGY = 'height_energy'
     TWO_DEGREES_OF_FREEDOM = '2DOF'
     # TODO these are a little out of place atm
-    SIMPLE = 'simple'
+    HEIGHT_ENERGY = 'height_energy'
```
Comment (jkirk5) [inline aviary/variable_info/enums.py:62]: You made them non-alphabetical now! Plus the TODO is implied to apply to height_energy now, which isn't accurate

**Review by jkirk5 — CHANGES_REQUESTED** [review]
Good stuff! I have a ton of little nitpicks about some changes, especially instances where simple was being brought back to name or label things.

**Comment by jkirk5** [comment]
> I am mostly thinking in terms of the MBSAE published work, and being able to reproduce it exactly as we migrate all of that to Aviary 0.9.x.

I don't think its reasonable to always expect identical results moving to a new version of Aviary, especially as the code is rapidly evolving. Long term best practice will be archiving the environment used to run final results in case replication is needed.

File: aviary/docs/getting_started/onboarding_level1.ipynb
Comment (johnjasa) [reply aviary/docs/getting_started/onboarding_level1.ipynb:1]: Thanks for your comments, Jason -- anything where simple snuck back in was just an artifact of this branch being out-of-date with main. I was waiting for Ken's assessment for MBSA&E work. I've worked through the merge conflicts now; I'll update any remaining notions of simple as you pointed out.

File: aviary/mission/flops_based/ode/mission_EOM.py (hunk 7 above)
Comment (johnjasa) [reply aviary/mission/flops_based/ode/mission_EOM.py:22]: changed!

File: aviary/mission/flops_based/ode/mission_ODE.py (hunk 8 above)
Comment (johnjasa) [reply aviary/mission/flops_based/ode/mission_ODE.py:42]: done!

File: aviary/models/test_aircraft/aircraft_for_bench_GwFm_simple.csv
Comment (johnjasa) [reply aviary/models/test_aircraft/aircraft_for_bench_GwFm_simple.csv:1]: removed

File: aviary/models/test_aircraft/aircraft_for_bench_FwFm_simple.csv
Comment (johnjasa) [reply aviary/models/test_aircraft/aircraft_for_bench_FwFm_simple.csv:1]: removed

File: aviary/validation_cases/benchmark_tests/test_subsystems_within_a_mission.py (hunk 11 above)
Comment (johnjasa) [reply aviary/validation_cases/benchmark_tests/test_subsystems_within_a_mission.py:24]: Changed for no particular reason; I copied-pasted this phase_info from another place. It is indeed mission agnostic

File: aviary/variable_info/enums.py (hunk 12 above)
Comment (johnjasa) [reply aviary/variable_info/enums.py:62]: order moved and comment removed

File: aviary/validation_cases/benchmark_tests/test_bench_FwFm.py [hunk 13]
```diff
@@ -340,32 +328,32 @@ def bench_test_swap_4_FwFm_simple(self):
...
                 "include_landing": True,
                 "constrain_range": True,
-                "target_range": (3360.0, "nmi"),
+                "target_range": (3368.0, "nmi"),
             },
         }
 
         prob = run_aviary(
-            'models/test_aircraft/aircraft_for_bench_FwFm_simple.csv', phase_info)
+            'models/test_aircraft/aircraft_for_bench_FwFm.csv', phase_info, max_iter=50)
```
Comment (Kenneth-T-Moore) [inline aviary/validation_cases/benchmark_tests/test_bench_FwFm.py:351]: Are we hitting maxiter on this benchmark? It is the FLOPS/FLOPS test, so I would expect it to converge more quickly.

File: aviary/validation_cases/benchmark_tests/test_0_iters.py [hunk 14]
```diff
@@ -63,6 +63,4 @@ def test_zero_iters_solved(self):
```
Comment (Kenneth-T-Moore) [inline aviary/validation_cases/benchmark_tests/test_0_iters.py:64]: Please keep these.

Note: testflo doesn't require `unittest.main()`.  We only need that if we run with unittest, or if we want the test for run with `python test_0_iters.py`.  You can't debug through unittest though, which is why I like to keep the other lines.

File: aviary/subsystems/propulsion/test/test_custom_engine_model.py [hunk 15]
```diff
@@ -126,67 +122,50 @@ def get_initial_guesses(self):
...
         prob = AviaryProblem(reports=False)
 
         # Load aircraft and options data from user
         # Allow for user overrides here
-        prob.load_inputs(csv_path, phase_info, engine_builder=SimpleTestEngine())
+        prob.load_inputs("models/test_aircraft/aircraft_for_bench_GwFm.csv",
+                         phase_info, engine_builder=SimpleTestEngine())
 
 
 # Preprocess inputs
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/propulsion/test/test_custom_engine_model.py:171]: indentation

File: aviary/subsystems/propulsion/test/test_custom_engine_model.py [hunk 16]
```diff
@@ -225,34 +204,6 @@ def test_custom_engine(self):
...
-            'traj.phases.cruise.indep_states.states:velocity',
-            'traj.phases.cruise.indep_states.states:mass',
-            'traj.phases.cruise.indep_states.states:distance',
-        ]
-
-        # Check that all expected variable names are present in design_vars
-        for var_name in expected_var_names:
-            self.assertIn(var_name, design_vars_dict)
-
-        # Check values
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/propulsion/test/test_custom_engine_model.py:245]: Why were these asserts all removed?

File: aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py [hunk 17]
```diff
@@ -279,6 +349,4 @@ def build_pre_mission(self, aviary_inputs):
 
 
 if __name__ == "__main__":
-    # unittest.main()
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py:282]: please restore these.

File: aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py [hunk 18]
```diff
@@ -91,6 +88,80 @@ def test_solved_aero_pass_polar(self):
...
+
+        prob.run_model()
+
+        CL_pass = prob.get_val("traj.cruise.rhs_all.core_aerodynamics.tabular_aero.CL")
+        CD_pass = prob.get_val("traj.cruise.rhs_all.core_aerodynamics.tabular_aero.CD")
+
+        assert_near_equal(CL_pass, CL_base, 1e-6)
+        assert_near_equal(CD_pass, CD_base, 1e-6)
+
+    def test_solved_aero_pass_polar_unique_abscissa(self):
```
Comment (Kenneth-T-Moore) [inline aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py:117]: Was this test deleted in a regression, or was it moved from elsewhere? I remember writing this, but it appears as new in this diff.

File: aviary/mission/flops_based/ode/mission_ODE.py [hunk 19]
```diff
@@ -152,3 +196,12 @@ def setup(self):
                                ('mass', Dynamic.Mission.MASS)
                            ],
                            promotes_outputs=['initial_mass_residual'])
+
+        self.nonlinear_solver = om.NewtonSolver(solve_subsystems=True,
```
Comment (Kenneth-T-Moore) [inline aviary/mission/flops_based/ode/mission_ODE.py:200]: Adding the solver has some ramifications. I wondered if you could confine it to a subsystem, but looks like that subsystem would need to contain at least the engine, and that's in the external subsystem group, so there probably isn't anything else we can do.

**Review by jkirk5 — APPROVED** [review]

File: aviary/interface/default_phase_info/height_energy.py [hunk 20]
```diff
@@ -17,188 +15,85 @@
...
-        'include_landing': True,
+    "post_mission": {
+        "include_landing": False,
+        "constrain_range": True,
+        "target_range": (1906, "nmi"),
     },
 }
-
-
-def phase_info_parameterization(phase_info, aviary_inputs):
```
Comment (Kenneth-T-Moore) [inline aviary/interface/default_phase_info/height_energy.py:126]: As in earlier comment, this needs to be migrated to not lose a command line feature that was essential to the release.

File: aviary/interface/methods_for_level2.py [hunk 21]
```diff
@@ -2416,30 +2383,35 @@ def _add_flops_landing_systems(self):
...
                            Mission.Landing.TOUCHDOWN_MASS, src_indices=[-1])
-        # TODO: approach velocity should likely be connected
-        if self.mission_method is HEIGHT_ENERGY:
-            self.model.connect('traj.descent.states:altitude', Mission.Landing.INITIAL_ALTITUDE,
-                               src_indices=[-1])
-        else:
-            self.model.connect('traj.descent.control_values:altitude', Mission.Landing.INITIAL_ALTITUDE,
-                               src_indices=[0])
+        self.model.connect('traj.descent.control_values:altitude', Mission.Landing.INITIAL_ALTITUDE,
+                           src_indices=[0])
```
Comment (Kenneth-T-Moore) [inline aviary/interface/methods_for_level2.py:2414]: Can you take a look at an n2 of one of the FLOPS models and make sure that none of the component changes are producing any cycles (i.e., everything is in the correct order)?

**Review by Kenneth-T-Moore — CHANGES_REQUESTED** [review]
A few comments to address.

File: aviary/validation_cases/benchmark_tests/test_bench_FwFm.py (hunk 13 above)
Comment (johnjasa) [reply aviary/validation_cases/benchmark_tests/test_bench_FwFm.py:351]: Not hitting max_iter; I added it during debugging

File: aviary/validation_cases/benchmark_tests/test_0_iters.py (hunk 14 above)
Comment (johnjasa) [reply aviary/validation_cases/benchmark_tests/test_0_iters.py:64]: Changed back!

File: aviary/subsystems/propulsion/test/test_custom_engine_model.py (hunk 15 above)
Comment (johnjasa) [reply aviary/subsystems/propulsion/test/test_custom_engine_model.py:171]: thanks, fixed

File: aviary/subsystems/propulsion/test/test_custom_engine_model.py (hunk 16 above)
Comment (johnjasa) [reply aviary/subsystems/propulsion/test/test_custom_engine_model.py:245]: Not important for testing if the custom engine model was used correctly; I had to update their names and realized there was no reason for them to be there

File: aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py (hunk 17 above)
Comment (johnjasa) [reply aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py:282]: restored

File: aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py (hunk 18 above)
Comment (johnjasa) [reply aviary/subsystems/aerodynamics/flops_based/test/test_solved_aero_group.py:117]: Great catch, it was unnecessarily duplicated in this PR, I fixed that now

File: aviary/interface/default_phase_info/height_energy.py (hunk 20 above)
Comment (johnjasa) [reply aviary/interface/default_phase_info/height_energy.py:126]: Agreed, added back now.

**Comment by johnjasa** [comment]
Thanks for these comments, @Kenneth-T-Moore! Please let me know if you'd like to see any other changes. I'm working on the docs failures now.

**Review by Kenneth-T-Moore — APPROVED** [review]

**Pull request merged**